File: php/src/Application/Catalogue/UomRepository.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

use Alxarafe\App\Domain\Catalogue\ValueObject\Uom;
use Alxarafe\App\Domain\Catalogue\ValueObject\UomCode;

interface UomRepository
{
    public function findByCode(UomCode $code): ?Uom;

    /** @return list<Uom> */
    public function findAll(): array;

    public function save(Uom $uom): void;
}

File: java/src/main/java/com/alxarafe/app/application/catalogue/UomRepository.java
package com.alxarafe.app.application.catalogue;

import com.alxarafe.app.domain.catalogue.valueobject.Uom;
import com.alxarafe.app.domain.catalogue.valueobject.UomCode;

import java.util.List;
import java.util.Optional;

public interface UomRepository {
    Optional<Uom> findByCode(UomCode code);

    List<Uom> findAll();

    void save(Uom uom);
}